crypto/src/hash/traits.rs

```diff
@@ -36,6 +37,12 @@ where
     }
 }
 
+/// Trait that converts itself into either field(s) or bits(s)
+/// for constructing the given [ROInput]
+pub trait RandomOraclePartialInput {
+    fn add_self_to(&self, input: &mut ROInput);
```

Comment:
The construction of this trait feels a bit strange to me. Since the `ROInput` is the thing being mutated it would be much more natural to have the trait implemented on that and be generic over what is being added. e.g. following the example of Vec::push

```rust
pub trait RandomOraclePartialInput<T> {
    fn append(&mut self, value: T);
}
```

An alternative would be to derive the[ `core::ops::Add`](https://doc.rust-lang.org/std/ops/trait.Add.html) trait which wouldn't mutate the ROInput but could return a new one with the value added.

I think either of these would make more sense and make it easier to collect the implementations in one place rather than scattering them around the codebase to where the things to be added are located.

Reply:
That's a great suggestion! Fixed as suggested, thanks!